burn-import/src/burn/node/base.rs

```diff
@@ -114,20 +108,17 @@ impl<PS: PrecisionSettings> Serialize for Node<PS> {
 }
 
 impl<PS: PrecisionSettings> Node<PS> {
-    pub fn name(&self) -> &str {
+    pub fn name(&self) -> String {
         match self {
-            Node::Add(_) => "add",
-            Node::Matmul(_) => "matmul",
-            Node::Constant(_) => "constant",
-            Node::Conv2d(_) => "conv2d",
-            Node::MaxPool2d(_) => "max_pool2d",
-            Node::Linear(_) => "linear",
-            Node::BatchNorm(_) => "batch_norm",
-            Node::ReLU(_) => "relu",
-            Node::Flatten(_) => "flatten",
-            Node::LogSoftmax(_) => "log_softmax",
-            Node::Equal(_) => "equal",
-            Node::Sigmoid(_) => "sigmoid",
+            Node::Add(_) => "add".to_string(),
+            Node::Matmul(_) => "matmul".to_string(),
+            Node::Constant(_) => "constant".to_string(),
+            Node::Conv2d(_) => "conv2d".to_string(),
+            Node::MaxPool2d(_) => "max_pool2d".to_string(),
+            Node::Linear(_) => "linear".to_string(),
+            Node::BatchNorm(_) => "batch_norm".to_string(),
+            Node::Equal(_) => "equal".to_string(),
+            Node::Unary(unary) => unary.name.to_string(),
```

Comment:
Instead of `String`, can we define an `enum` for all `unary` operators? In this specific case we would have a double `match`, the second one after `Node::unary(node)`

Reply:
This is only used in logging, so the important part is having something clean, we already have an enum, but the `Default` formating isn't pretty enough for printing. Maybe this should be refactoring into a `Display` function instead of a "name".

Reply:
Yeah, that could be an idea

Reply:
@nathanielsimard @Luni-4 

There is a [`strum_macros::Display`
](https://docs.rs/strum_macros/0.25.1/strum_macros/derive.Display.html) derive that you can use. You can also specify serialization types, see https://docs.rs/strum/latest/strum/additional_attributes/index.html. We will basically use this:

```rust
#[derive(Debug, strum_macros::Display)]
#[strum(serialize_all = "snake_case")]
```

We are using strum in burn-import.

Reply:
@antimora  @nathanielsimard 

For now we can use this approach since it is already there, but I think we should try to use the less amount of dependencies as possible in my opinion. (when it's possible obviously)

Reply:
I agree but we are already using this dependency in burn-import.